commit 7058ed6465daa7026f1e34a43e8c5cda6adb21c2
Author: Po-Hsien Wang <pwang@chromium.org>
Date:   Tue Aug 8 15:51:05 2017 -0700

    Waive certain tests

    1) Add ability to waive certain tests in webgl_performance testing.
    2) Waive rgb float test as it is supported in the spec.

    BUG=chromium:753229
    TEST=test_that

    Change-Id: I1fb3c7af4894321a14ae2e9272fcc80280d6a410

diff --git a/index.html b/index.html
index 75c096d9b..7a5b5002d 100644
--- a/index.html
+++ b/index.html
@@ -6,6 +6,8 @@
 <script type="text/javascript" src="../js/speedtests.js"></script>
 <script>
   var testPageURLs = [];
+  var waivedtestPageURL = ['convert-Canvas-to-rgb-float.html',
+                           'convert-Canvas-to-rgb-float-premultiplied.html'];
   var testsRun = [];
   var currentPageIndex;
   var testframe;
@@ -17,6 +19,9 @@
   var test_report = "Tests did not complete.";
   var time_ms_geom_mean = -1.0;
   var test_completed = false;
+  var numberOfWaived = 0;
+  var numberOfPassed = 0;
+
   var rawResults = {};
   var useiframe = document.location.hash.indexOf("useWindow") == -1;
   var testwindow;
@@ -87,9 +92,14 @@
   function messageListener(e) {
     if (e.data.testDescription) {
       testsRun.push(e.data);
+
+      if (waivedtestPageURL.indexOf(testPageURLs[currentPageIndex]) >= 0){
+        numberOfWaived++;
+      }
-      if (e.data.testResult) {
+      else if (e.data.testResult) {
         sumOfResults += e.data.testResult;
         sumOfLogResults += Math.log(e.data.testResult);
+        numberOfPassed++;
       }
       numberOfResults++;
       currentPageIndex++;
